tests/beaker_tests/snmp_notification_receiver/snmp_notification_receiver_provider_defaults.rb

```diff
@@ -90,8 +92,13 @@
     on(agent, cmd_str) do
       search_pattern_in_output(stdout, { 'ensure' => 'present' },
                                false, self, logger)
-      search_pattern_in_output(stdout, { 'source_interface' => 'ethernet1/3' },
-                               false, self, logger)
+      if operating_system == 'ios_xr'
+        search_pattern_in_output(stdout, { 'source_interface' => 'gigabitethernet0/0/0/0' },
```

Comment:
We have a method `find_interface` that mitigates hard coding of interface names:
- https://github.com/cisco/cisco-network-puppet-module/blob/175e4d68f93f0da696a2df346a88a6156704cff0/tests/beaker_tests/cisco_interface/test_interface_vlan_mapping.rb#L65
- https://github.com/cisco/cisco-network-puppet-module/blob/8f2b571919e65d27e588fcd9c5862e6da083d3f9/tests/beaker_tests/ntp_config/ntp_config_provider_defaults.rb#L86